library/x509_crt.c

```diff
@@ -700,11 +703,8 @@ static int x509_get_subject_alt_name( unsigned char **p,
             cur = cur->next;
         }
 
-        buf = &(cur->buf);
-        buf->tag = tag;
-        buf->p = *p;
-        buf->len = tag_len;
-        *p += buf->len;
+        cur->buf = tmp_san_buf;
```

Comment:
Although technically this works, since it is assigning by value `tmp_buf` to `cur->buf`, I don't like that there is an assignment of a structure defined in the stack, without a proper memory copying.

Reply:
Yes, that's deliberate - why do you prefer a `memcpy()`?

Reply:
Since you are counting for the struct to be copied by value, it wouldn't do a deep-copy(neither does `memcpy` actually), and this is not future proof.
In addition, in a matter of style, I always prefer using to copy data from non primitive types.